app/views/devise/confirmations/new.html.haml

```diff
@@ -3,7 +3,7 @@
   = devise_error_messages!
   .field
     %h4= f.label :email
-    = f.email_field :email, class: 'form-control', autofocus: true, value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email)
+    = f.email_field :email, class: 'form-control', autofocus: true, required: true, value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email)
```

Comment:
Advice: it's always are to read a long line of key/value arguments. Feel free to split it.
In this case, I would probably write something like this:

```ruby
f.email_field :email,
              class: 'form-control',
              autofocus: true,
              required: true,
              value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email)
```